ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerSession.scala

```diff
@@ -3,38 +3,45 @@
 
 package com.daml.ledger.api.testtool.infrastructure
 
-import com.daml.ledger.api.testtool.infrastructure.participant.ParticipantSessionManager
+import com.daml.ledger.api.testtool.infrastructure.participant.{
+  ParticipantSession,
+  ParticipantSessionManager
+}
 
+import scala.collection.immutable
 import scala.concurrent.{ExecutionContext, Future}
 
-private[testtool] final class LedgerSession(
-    val config: LedgerSessionConfiguration,
-    participantSessionManager: ParticipantSessionManager,
+private[infrastructure] final class LedgerSession(
+    shuffleParticipants: Boolean,
+    participantSessions: immutable.Seq[(String, ParticipantSession)],
 )(implicit val executionContext: ExecutionContext) {
-
-  private[this] val endpointIdProvider =
-    Identification.circularWithIndex(Identification.greekAlphabet)
-
-  private[this] val participantSessions =
-    Future
-      .sequence(config.participants.map(hostAndPort =>
-        participantSessionManager.getOrCreate(config.forParticipant(hostAndPort))))
-      .map(_.map(endpointIdProvider() -> _))
-
-  private[testtool] def createTestContext(
+  private[infrastructure] def createTestContext(
       applicationId: String,
       identifierSuffix: String,
   ): Future[LedgerTestContext] =
-    participantSessions.flatMap { sessions =>
-      Future
-        .sequence(
-          (if (config.shuffleParticipants) scala.util.Random.shuffle(sessions) else sessions)
-            .map {
-              case (endpointId, session) =>
-                session.createTestContext(endpointId, applicationId, identifierSuffix)
-            }
-        )
-        .map(new LedgerTestContext(_))
-    }
+    Future
+      .sequence(
+        (if (shuffleParticipants) scala.util.Random.shuffle(participantSessions)
+         else participantSessions)
+          .map {
+            case (endpointId, session) =>
+              session.createTestContext(endpointId, applicationId, identifierSuffix)
+          }
+      )
+      .map(new LedgerTestContext(_))
+}
 
+object LedgerSession {
+  def apply(
+      config: LedgerSessionConfiguration,
```

Comment:
Why are you taking the whole configuration?

Reply:
Good point. Fixed.